### ISSUE TITLE
[BUG] NET ERROR: index 1024 is out of bounds for axis 1 with size 1024

### ISSUE THREAD

**Issue by FranJGonCal** [MouseLand/cellpose#1004]
Hello, I'm new to cellpose and I find an error when upload my file to the program.
When I upload an RGB image and run the cyto3 model to segment the image (to later train a custom model) using the gui, an error NET ERROR: index 1024 is out of bounds for axis 1 with size 1024 appears, so no mask are found, and also no segmentation is performed.
I don't really know if I need to manually draw the mask of the cells that I want to segment before running the cyto3 model.

<img width="563" alt="Captura de pantalla 2024-08-29 a las 12 35 43" src="https://github.com/user-attachments/assets/eb968582-aecf-4814-9840-b8e3229c92e7">

**Comment by carsen-stringer** [comment]
is this image a z-stack or a single image? does it show up correctly in the GUI? if you have a z-stack then you want to open cellpose with `python -m cellpose --Zstack`

**Comment by FranJGonCal** [comment]
> is this image a z-stack or a single image? does it show up correctly in the GUI? if you have a z-stack then you want to open cellpose with `python -m cellpose --Zstack`

It's a single image.

Trying different approaches, I found a way in which It's run correctly. When uploading the image to the program, I was using the "use GPU" option and the problem which I was commenting appeared. However, if I do not choose this option, the progrma runs correctly.

Anyway, thanks so much for your answer.

**Comment by carsen-stringer** [comment]
Weird thanks for the update, somehow the GPU torch version isn't work, I would also try updating to the latest version v3.0.11 (`pip install cellpose --upgrade`) and upgrading torch in the same way

**Comment by tathey1** [comment]
Having the same issue on MacOS, cellpose 3.0.11, 2D images (IndexError in line 194 `dynamics.py`). However the issue went away when I reverted using `pip install cellpose==3.0.8`. Happy to share more info if useful.

**Comment by carsen-stringer** [comment]
hmm nothing in that function has changed in the past 7 months. 

can you upgrade to the latest github version? `pip install git+https://github.com/mouseland/cellpose.git`

if that doesn't work, could you please check if masks_padded is a reasonable size? https://github.com/MouseLand/cellpose/blob/52f75f9636250979dcff48da18d24aaff41c2c86/cellpose/dynamics.py#L159

**Comment by tathey1** [comment]
After poking around a little, I think your suggestion to upgrade torch was right. I was in a Python 3.9.5 environment and for some reason torch 2.2 was the newest version available. Now I'm in a Python 3.9.6 environment which allowed me to get torch 2.5 and I now longer get the error.

To anyone it may concern: with the older torch version, the traceback was showing an Index Error [here](https://github.com/MouseLand/cellpose/blob/52f75f9636250979dcff48da18d24aaff41c2c86/cellpose/dynamics.py#L194). It might have to do with function `torch.nonzero` with the mps backend because when I used `device = torch.device("cpu")` earlier in the function, the issue was resolved. Thanks!
